__init__.py

```diff
@@ -733,27 +786,27 @@ def spotify_play(self, dev_id, uris=None, context_uri=None):
             self.dev_id = dev_id
         except spotipy.SpotifyException as e:
             # TODO: Catch other conditions?
-            self.speak_dialog('NotAuthorized')
+            self.speak_dialog('NotConnected')
         except Exception as e:
-            LOG.exception(e)
-            self.speak_dialog('NotAuthorized')
+            self.handle_error(e)
 
     def start_playlist_playback(self, dev, name, uri):
-        if dev and uri:
-            LOG.info(u'playing {} using {}'.format(name, dev['name']))
-            self.speak_dialog('ListeningToPlaylist',
-                              data={'playlist': name})
-            time.sleep(2)
-            tracks = self.spotify.user_playlist_tracks(uri['owner']['id'],
-                                                       uri['id'])
-            uris = [t['track']['uri'] for t in tracks['items']]
-            self.spotify_play(dev['id'], uris=uris)
-            return True
-        elif not dev:
-            LOG.info('No spotify devices found')
-        else:
-            LOG.info('No playlist found')
-        return False
+        if not dev:
+            self.handle_error('No spotify devices found')
+            return False
```

Comment:
I don't really like this scheme where error handling takes precedence over the actual logic, I like to have the important things first and then the error conditions are handled. To me that makes more sense.

Reply:
It does not really take precedence. The following is logically equivalent.
```
if condition:
  do_something()
else:
  do_report()
```
```
if not condition:
  do_report()
else:
  do_something()
```
```
if not condition:
  do_report()
  return
do_something()
```
It does however make the code more readable when the logic is a bit longer. It also ensures that abnormal conditions are handled, which is often forgotten. And prevents for nesting of conditions.
```
if condition:
  if other_confition:
    logic_1()
    logic_2()
    logic_3()
    ...
    ...
    ...
  else:
    do_report()
else:
  do_report()
```
```
if not condition:
  do_report()
  return
if not other_confition:
  do_report()
  return
logic_1()
logic_2()
logic_3()
...
...
...
```

Reply:
I understand what you mean I'm not sure I'm 100% on board with it and I think I can do some reorganization to make this a bit simpler...fewer points that need logging